packages/react-native/React/Views/ScrollView/RCTScrollView.m

```diff
@@ -324,7 +325,20 @@ - (void)_keyboardWillChangeFrame:(NSNotification *)notification
   }
 
   CGPoint newContentOffset = _scrollView.contentOffset;
-  CGFloat contentDiff = endFrame.origin.y - beginFrame.origin.y;
+  self.firstResponderFocus = CGRectNull;
+
+  CGFloat contentDiff = 0;
+  if ([[UIApplication sharedApplication] sendAction:@selector(reactUpdateResponderOffsetForScrollView:) to:nil from:self forEvent:nil]) {
+    // Inner text field focused
+    CGFloat focusEnd = self.firstResponderFocus.origin.y + self.firstResponderFocus.size.height;
```

Comment:
```suggestion
    CGFloat focusEnd = CGRectGetMaxY(self.firstResponderFocus);
```

Reply:
This code currently assumes the scrollview is vertical, it should really work when content is laid out horizontally too.

Reply:
@javache This `_keyboardWillChangeFrame:` method has an existing `if ([self isHorizontal:_scrollView]) { return; }` check at the top so this code only applies to vertical scroll views. Fortunately when the scroll view is horizontal the default behaviour correctly scrolls partially-obscured text fields into view when focused.